### PR TITLE
Fix typo in skipTestsFor330 calculation [skip ci]

### PR DESCRIPTION
**Pull request by gerashegalov**
- Allow overriding SPARK_SHIM_VERSIONS from Jenkins for faster smoke tests
- add required params to deploy-file: -Durl, -DrepositoryId 

Addendum for #4355 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

File: jenkins/spark-nightly-build.sh [hunk 1]
```diff
@@ -49,6 +50,7 @@ function distWithReducedPom {
 
         deploy)
             mvnCmd="deploy:deploy-file"
+            mvnExtaFlags="-Durl=$SERVER_URL -DrepositoryId=$SERVER_ID"
```
Comment (NvTimLiu) [inline jenkins/spark-nightly-build.sh:53]: @Gera Shegalov  We already have this URL defined in Jenkins as below instead, similar referrence in [settings.xml](https://github.com/NVIDIA/spark-rapids/blob/branch-22.02/jenkins/settings.xml#L105)

`-Durl=${URM_URL}-local -DrepositoryId=snapshots`

just change to
`mvnExtaFlags="-Durl=${URM_URL}-local -DrepositoryId=snapshots"`
from
`mvnExtaFlags="-Durl=$SERVER_URL -DrepositoryId=$SERVER_ID"`
BTW, I did not found where this `mvnExtaFlags`  been used in the deloy:deploy-file CLI in PR4371?

File: jenkins/spark-nightly-build.sh [hunk 2]
```diff
@@ -89,7 +91,7 @@ for buildver in "${SPARK_SHIM_VERSIONS[@]:1}"; do
 done
 
 mvn -B clean install -pl '!tools' \
-    -PsnapshotsWithDatabricks \
+    $DIST_PROFILE_OPT \
```
Comment (tgravescs) [inline jenkins/spark-nightly-build.sh:94]: this isn't pulling the databricks jars is it?

SPARK_SHIM_VERSIONS_STR=${SPARK_SHIM_VERSIONS_STR:-"301 302 303 304 311 311cdh 312 313 320 321 330"}

I don't see DB in there and the nightly needs to publish the jar with databricks

**Review by tgravescs — DISMISSED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
Verified the pom for dist is now correct for snapshots as well

**Pull request merged**